packages/mongodb-chatbot-server/src/routes/responses/createResponse.ts

```diff
@@ -491,10 +505,14 @@ const convertInputToDBMessages = (
   }
 
   return input.map((message) => {
-    // handle function tool calls and outputs
-    const role = message.type === "message" ? message.role : "system";
-    const content =
-      message.type === "message" ? message.content : message.type ?? "";
+    // set default role and content for function tool calls and outputs
+    let role: MessagesParam[number]["role"] = "system";
```

Comment:
i am almost certain this is wrong. role. i think role should be 'tool' or 'assistant'. can do a follow up ticket to address since looks like bug preceded this work

Reply:
Yeah, gonna make a followup for this